utils/buttons.py

```diff
@@ -542,12 +594,16 @@ async def _get_kwargs_from_page(self, page: Any) -> Dict[str, Any]:
     def format_view(self) -> None:
         for i, b in enumerate(self.children):
             b.disabled = any(
-                [self.current_page == 0 and i < 2, self.current_page == self._source.get_max_pages() - 1 and not i < 3]
+                [
+                    self.current_page == 0 and i < 2,
+                    self.current_page == self._source.get_max_pages() - 1 and i > 2 and not b.stay_active
+                ]
             )
 
     async def interaction_check(self, interaction: discord.Interaction) -> bool:
         """Only allowing the context author to interact with the view"""
         ctx = self.ctx
+        assert ctx is not None
```

Comment:
wtf no assert allowed in stella bot

Reply:
due to current design this assert fixes like 7 type errors below. either this assert stays or there will have to be a lot of `type: ignore` comments or design change is needed